src/main/java/net/md_5/itag/iTag.java

```diff
@@ -46,15 +51,27 @@ public void onEnable()
         }
 
         getServer().getPluginManager().registerEvents( this, this );
-        ProtocolLibrary.getProtocolManager().addPacketListener( new PacketAdapter( this, PacketType.Play.Server.NAMED_ENTITY_SPAWN )
+        ProtocolLibrary.getProtocolManager().addPacketListener( new PacketAdapter( this, PacketType.Play.Server.PLAYER_INFO )
         {
             @Override
             public void onPacketSending(PacketEvent event)
             {
-                event.getPacket().getGameProfiles().write( 0, getSentName( event.getPacket().getIntegers().read( 0 ), event.getPacket().getGameProfiles().read( 0 ), event.getPlayer() ) );
+                if (event.getPacket().getPlayerInfoAction().read(0) != PlayerInfoAction.ADD_PLAYER) return;
+		List<PlayerInfoData> newPlayerInfoDataList = new ArrayList<>();	
+		List<PlayerInfoData> playerInfoDataList = event.getPacket().getPlayerInfoDataLists().read(0);
+		for (PlayerInfoData playerInfoData : playerInfoDataList) {
+		    if (playerInfoData == null || playerInfoData.getProfile() == null || Bukkit.getPlayer(playerInfoData.getProfile().getUUID()) == null) newPlayerInfoDataList.add(playerInfoData); //Unknown Player
```

Comment:
You call getPlayer twice, here and in the getEntityId method, just store the bukkit player in the loop